src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalResult = [];
+  const cloneCopy = { ...state };
+
+  actions.forEach((el) => {
+    switch (el.type) {
+      case 'addProperties': {
+        for (const keys in el.extraData) {
+          cloneCopy[keys] = el.extraData[keys];
+        }
+
+        finalResult.push({ ...cloneCopy });
```

Comment:
[checklist:](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) "If you perform same action in all switch cases - do it just once afterwards"